source/agora/utils/Log.d

```diff
@@ -115,3 +115,43 @@ public class AgoraLayout : Appender.Layout
         }
     }
 }
+
+/***************************************************************************
+
+    Logging function which is only called from C++ code
+
+    It's for C++ code to use agora's logger instead of C++ stdout 
+
+    Params:
+        logger = the logger name
+        level = the logging level
+        msg = the log message
+
+***************************************************************************/
+private extern (C++) void DLogging (const char* logger, int level, const char* msg)
+{
+    import std.string;
+
+    auto log = Log.lookup(fromStringz(logger));
+    if (log !is null)
```

Comment:
In D you can do the following:
```suggestion
if (auto log = Log.lookup(fromStringz(logger))
```

However, `Log.lookup` should never return `null`, so I would say don't do the check, or make it an `assert`.
Additionally, I would check the value of `level` before casting.